types/artifacts.d.ts

```diff
@@ -318,13 +318,18 @@ declare global {
       /** @see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#Attributes */
       export interface AnchorElement {
         rel: string
+        /** The computed href property: https://www.w3.org/TR/DOM-Level-2-HTML/html.html#ID-88517319, use `rawHref` for the exact attribute value */
         href: string
+        /** The exact value of the href attribute value, as it is in the DOM */
+        rawHref: string
+        name?: string
         text: string
         target: string
         devtoolsNodePath: string
         selector: string
         nodeLabel: string
         outerHTML: string
+        listeners?: Array<Crdp.DOMDebugger.EventListener>
```

Comment:
since this is a blessed artifact we promise the API to plugins won't break we might want to be a little more selective here than returning the entire Crdp value that *isn't* as fixed

perhaps we can just return the `type` and the stringified handler function if it's serializable?

Reply:
> perhaps we can just return the `type` and the stringified handler function if it's serializable?

if we reallllllly want to do the handler source check, this might be worth breaking out into a specialized gatherer that doesn't keep the full source around. I hacked on a script serializer to this PR and it was +5MB of event listener source in the artifacts for https://www.google.com/search?q=yah all to search for `window.opener` or `location` :)

(weirdly it's bigger than `ScriptElements` + `MainDocumentContent` for that same page, which is fun :)

Reply:
> (weirdly it's bigger than ScriptElements + MainDocumentContent for that same page, which is fun :)

oh, mystery looks like it may be there's only two listeners, a ~600KB one and a ~450KB one, but they're bound multiple times each. `eval\(\w\)` does match in there twice, though 😬

Reply:
how in the world is it so big 😱 

Linking these up is going to be an enormous pain though, and I'm not sure it's worth separating. It feels like we might need to just break the norm here and have a `isSuspectedLinkHandler` or something on each listener. I know it's a bummer to have any audit logic leak into a generic gatherer, but the alternative appears to be *duplicating* the anchor elements gatherer just to add this boolean on a dedicated gatherer. Or did you have something else in mind @brendankenny ?

My other proposal which I think I prefer a little more is to only include the first X characters of the stringified handler. We're specifically looking for the simple link handler case and if it's much larger than that it's safe to assume it's a complicated button or something.

Reply:
> Linking these up is going to be an enormous pain though, and I'm not sure it's worth separating

Good point, I hadn't thought about that. Not fun :)

I'd be ok with the first idea, lightweight trade-off against artifact purity. The second one I feel like I'd want to run some more tests before saying. If for a lot of sites this would help it's all the beginning of the same string over and over (from one handler bound many times) it's a lot of bytes for something no other audits may ever use.

Reply:
alright @umaar I think we reached a consensus and want to expose just two properties on these listeners for now

```
interface AnchorElementListener {
  type: Crdp.DOMDebugger.EventListener['type']
  isSuspectedLinkHandler: boolean
}
```

`isSuspectedLinkHandler` for now just checks the stringification of the listener to see if it looks like a window.location/window.open situation.

Reply:
I guess we've slightly changed this now, I think because the convo has carried on later on this thread, this could be resolved maybe

Reply:
ah yeah we have shifted a bit since then. let's start fresh. 👍